### PR TITLE
Adding unregister method

### PR DESCRIPTION
**Pull request by matheuscampanhaf**
Adding method to unregister a callback.
Now, messenger returns a callbackId that can be used to unregister the callback.

### CODE REVIEW

File: lib/messenger.js [hunk 1]
```diff
@@ -263,11 +264,40 @@ class Messenger {
...
+      identifier = callbackId;
+
+    }
+    this.eventCallbacks[subject][event][identifier] = callback
 
     if (!(subject in this.subjects) && !(subject in this.globalSubjects)) {
       this.createChannel(subject);
     }
+
+    if(!callbackId){
```
Comment (giovannicuriel) [inline lib/messenger.js]: How about it always return an ID, regardless whether it was set by the user or not? Therefore, the user can always retrieve an ID when calling this function.

File: index.d.ts [hunk 2]
```diff
@@ -2,11 +2,12 @@ declare module '@dojot/dojot-module' {
     class Messenger {
         public constructor(name: string, config?: any);
         public emit(subject: string, tenant: string, event: string, data: any): void;
-        public on(subject: string, event: string, callback: (tenant: string, data: any) => void): void;
+        public on(subject: string, event: string, callback: (tenant: string, data: any) => void, callbackId?: string): any;
```
Comment (giovannicuriel) [inline index.d.ts]: The return here is always a string, right?

File: lib/messenger.js [hunk 3]
```diff
@@ -253,21 +255,50 @@ class Messenger {
    * @param {string} event
    * @param {function} callback
    */
-  on(subject, event, callback) {
+  on(subject, event, callback, callbackId = false) {
```
Comment (giovannicuriel) [inline lib/messenger.js]: I think that we could use `= undefined` here, but I'm not sure.

File: lib/messenger.js [hunk 4]
```diff
@@ -253,21 +255,50 @@ class Messenger {
...
-      this.eventCallbacks[subject][event] = [];
+      this.eventCallbacks[subject][event] = {};
     }
 
-    this.eventCallbacks[subject][event].push(callback);
+    let identifier = uuid4();
+    if(callbackId){
+      identifier = callbackId;
+
+    }
```
Comment (giovannicuriel) [inline lib/messenger.js]: How about
```
if (callbackId) { 
   identifier = callbackId;
} else {
  identifier = uuid4()
} 
```

Comment (matheuscampanhaf) [reply lib/messenger.js]: Seems better!

File: lib/messenger.js (hunk 1 above)
Comment (matheuscampanhaf) [reply lib/messenger.js]: Seems better.

File: index.d.ts (hunk 2 above)
Comment (matheuscampanhaf) [reply index.d.ts]: With the changes above, yes!

**Pull request merged**